src/lib/input/input.ts

```diff
@@ -262,7 +267,19 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
     }
 
     this._isServer = !this._platform.isBrowser;
-    this._isNativeSelect = this._elementRef.nativeElement.nodeName.toLowerCase() === 'select';
+    this._isNativeSelect = element.nodeName.toLowerCase() === 'select';
+
+    // Add a class to the form field when it has a
+    // native select so we can add some specific styling.
+    if (formField && this._isNativeSelect) {
```

Comment:
`mat-form-field` has built in support for this by setting the `controlType`. Above it is set to `mat-input`, instead we can conditionally set it to `mat-native-select` or `mat-native-select-multiple`